routes/soldiers.js

```diff
@@ -0,0 +1,30 @@
+const soldiers = require('../collections/soldiers.js');
+const urlib = require('url');
+
+function handleSoldierReq(req, soldierData, done) {
+
+  const parsedUrlString = req.url.split("/")
+  if (parsedUrlString.length > 3) {
+    done(new Error())
+  }
+  const soldierId = parsedUrlString[2];
+  const parsedUrl = new urlib.URL("http://localhost:3000/" + req.url);
+  if (req.method === "GET") {
+    if (parsedUrlString.length === 3 && /tt[0-9]{7}/.test(soldierId)) {
```

Comment:
Notice that's you're mixing 2 distinct things in here, checking which route was called(your first check) & validating that the input is in the right format.
You need to separate them because you wouldn't want someone to receive 404 when he simply wrote the wrong soldier id.